### PR TITLE
Task #196315 : Adding Table along with API call

### PR DESCRIPTION
**Pull request by Xitija**
Created a table for showing users of organization when organization Id is passed to API and displaying them on UI

### CODE REVIEW

File: projects/sb-user-management/src/lib/user-list/user-list.component.ts [hunk 1]
```diff
@@ -1,15 +1,384 @@
...
   templateUrl: './user-list.component.html',
   styleUrls: ['./user-list.component.css']
 })
 export class UserListComponent implements OnInit {
+  tableColumns: any[] = [];;
+  users : any ;
+  constructor() {}
+
+  // for time being dummy data
+  students = [
```
Comment (gouravmore) [inline projects/sb-user-management/src/lib/user-list/user-list.component.ts]: @Xitija Please remove these hardcoded values and do the actual API call for user list

File: projects/sb-user-management/src/lib/user-list/user-list.component.ts [hunk 2]
```diff
@@ -1,15 +1,384 @@
 import { Component, OnInit } from '@angular/core';
-
+import { BrowserModule } from '@angular/platform-browser';
 @Component({
   selector: 'lib-user-list',
   templateUrl: './user-list.component.html',
   styleUrls: ['./user-list.component.css']
 })
 export class UserListComponent implements OnInit {
+  tableColumns: any[] = [];;
```
Comment (gouravmore) [inline projects/sb-user-management/src/lib/user-list/user-list.component.ts]: remove this extra semicolon

File: proxy.conf.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const PROXY_CONFIG = [
+    {
+        context: [
+            "/content-plugins",
```
Comment (gouravmore) [inline proxy.conf.js]: remove unwanted endpoints

**Review by gouravmore — CHANGES_REQUESTED** [review]

**Review by gouravmore — APPROVED** [review]

File: proxy.conf.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    {
+        context: [
+            "/action",
+            "/content",
+            "/api",
+            "/assets",
+            "/search"
+
+        ],
+        "target": "https://sunbirdsaas.com",
```
Comment (manojLondhe) [inline proxy.conf.js:11]: should we have config files for dev/uat/prod? and use those? @gouravmore

File: projects/sb-user-management/src/lib/user-list/user-list.component.ts [hunk 5]
```diff
@@ -1,15 +1,36 @@
...
+      { field: "lastName", header: "Last Name" },
+      //{ field: 'role', header: 'Role' },
+      { field: "email", header: "Email" },
+      { field: "phone", header: "Phone" },
+      { field: "status", header: "Status" },
+      { field: "channel", header: "Channel" },
+    ];
 
-  ngOnInit(): void {
+    this.userService.getUsersInOrganization("0124487522476933120").subscribe(
```
Comment (manojLondhe) [inline projects/sb-user-management/src/lib/user-list/user-list.component.ts:26]: what is 0124487522476933120 ?

File: projects/sb-user-management/src/lib/user-list/user-list.component.ts [hunk 6]
```diff
@@ -1,15 +1,36 @@
...
 export class UserListComponent implements OnInit {
+  tableColumns: any[] = [];
+  users: any;
+  count: any;
+  constructor(private userService: UserService) {}
 
-  constructor() { }
+  ngOnInit() {
+    this.tableColumns = [
+      { field: "firstName", header: "First Name" },
```
Comment (manojLondhe) [inline projects/sb-user-management/src/lib/user-list/user-list.component.ts:17]: can we use i18 lang constants and files for UI labels?

File: projects/sb-user-management/src/lib/user-list/user-list.component.html [hunk 7]
```diff
@@ -1 +1,42 @@
-<p>user-list works!</p>
+<div class="card">
+  <div class="flex align-items-center justify-content-between">
+    <h3>List of Users</h3>
```
Comment (manojLondhe) [inline projects/sb-user-management/src/lib/user-list/user-list.component.html:3]: can we use i18 lang constants and files for all UI labels?

File: projects/sb-user-management/src/lib/services/user-service/user.service.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+          "roles",
+          "managedBy",
+          "status",
+          "channel",
+        ],
+        limit: 50,
+      },
+    };
+
+    return this.http.post("api/user/v1/search", body, { headers: header });
```
Comment (manojLondhe) [inline projects/sb-user-management/src/lib/services/user-service/user.service.ts:39]: should endpoint URLs come from user-mgmt modules' config file'?

File: proxy.conf.js (hunk 4 above)
Comment (gouravmore) [reply proxy.conf.js:11]: Yes, we are doing the changes for different env files.

File: projects/sb-user-management/src/lib/user-list/user-list.component.ts (hunk 5 above)
Comment (gouravmore) [reply projects/sb-user-management/src/lib/user-list/user-list.component.ts:26]: Currently, we are hardcoding the organization value to get the users list.

File: projects/sb-user-management/src/lib/user-list/user-list.component.ts (hunk 6 above)
Comment (gouravmore) [reply projects/sb-user-management/src/lib/user-list/user-list.component.ts:17]: Yes, we can use it. Added this action item to the task list.

File: projects/sb-user-management/src/lib/services/user-service/user.service.ts (hunk 8 above)
Comment (gouravmore) [reply projects/sb-user-management/src/lib/services/user-service/user.service.ts:39]: Yes, this should come from urlConfig file.

**Comment by manojLondhe** [comment]
Closing unmerged

**Pull request closed without merging**